Ubuntu: add 24.04

moto-timo opened the pull request:
https://releases.ubuntu.com/noble/

                           RELEASED  END OF STANDARD SUPPORT
24.04 LTS (Noble Numbat)   Apr 2024  Apr 2034

https://ubuntu.com/about/release-cycle

https://canonical.com/blog/canonical-releases-ubuntu-24-04-noble-numbat

moto-timo committed:
Ubuntu: add 24.04

https://releases.ubuntu.com/noble/

                           RELEASED  END OF STANDARD SUPPORT
24.04 LTS (Noble Numbat)   Apr 2024  Apr 2034

https://ubuntu.com/about/release-cycle

https://canonical.com/blog/canonical-releases-ubuntu-24-04-noble-numbat

Signed-off-by: Tim Orling <tim.orling@konsulko.com>

Ecordonnier commented:
Is something missing before this can be merged?

Ecordonnier commented on dockerfiles/ubuntu/ubuntu-24.04/ubuntu-24.04-base/Dockerfile:
```diff
@@ -0,0 +1,72 @@
+    DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        gawk \
+        wget \
+        git-core \
```
git-core has been renamed to git a long time ago. "git-core" is provided as virtual package only for backwards compatibility.

moto-timo commented:
> Is something missing before this can be merged?

It failed to build
`2.313 E: Unable to locate package libegl1-mesa `

and the Yocto Project AutoBuilder was also having issues with Ubuntu 24.04 (notice there is no 24.04 [worker](https://autobuilder.yoctoproject.org/typhoon/#/workers), for good reason). I only have limited time to spend on this, and if it does not pass automation it will not be merged.

moto-timo committed:
centos: drop centos-7 EOL

https://www.redhat.com/en/blog/centos-linux-has-reached-its-end-life-eol

The end of life for CentOS Linux 7 has come and gone - on June 30, 2024.

Signed-off-by: Tim Orling <tim.orling@konsulko.com>

moto-timo committed:
fedora: drop EOL 36,37,38

https://docs.fedoraproject.org/en-US/releases/eol/

Release            EOL since
Fedora Linux 38    2024-05-21
Fedora Linux 37    2023-12-05
Fedora Linux 36    2023-05-16

Signed-off-by: Tim Orling <tim.orling@konsulko.com>

Ecordonnier commented:
> > Is something missing before this can be merged?
> 
> It failed to build `2.313 E: Unable to locate package libegl1-mesa `
> 
> and the Yocto Project AutoBuilder was also having issues with Ubuntu 24.04 (notice there is no 24.04 [worker](https://autobuilder.yoctoproject.org/typhoon/#/workers), for good reason). I only have limited time to spend on this, and if it does not pass automation it will not be merged.

Thanks, I was not trying to push you, but rather genuinely trying to understand if something was blocking this.
I'll have a look at the replacement for libegl1-mesa. Also the 24.04 image will need an apparmor profile for bitbake as documented in https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/2056555/comments/11

Ecordonnier commented:
In Ubuntu 22.04, libegl1-mesa was a transitional dummy package whose only dependency was libegl1, thus libegl1 is the correct replacement for libegl1-mesa in Ubuntu 24.04:
https://packages.ubuntu.com/jammy/libegl1-mesa

Ecordonnier commented:
Also note that libegl1 was removed completely from the ubuntu essential packages, so I think not even libegl1 is needed, and libegl1-mesa can be removed completely: https://git.yoctoproject.org/poky/commit/?id=fd1ea6a600a3b069d6cea4153663ea9720d4eafd 

> drop mesa/sdl from essential host packages
>    
>   They used to be required for qemu graphics support, but neither is being
>  for anything anymore.

moto-timo committed:
ubuntu-24.04: drop libegl1-mesa, libsdl1.2-dev

UBUNTU_HOST_PACKAGES_ESSENTIAL dropped libegl1-mesa libsdl1.2-dev in:
https://git.yoctoproject.org/yocto-docs/commit/?id=96e96382cecb4928f49d37bf870335fa6aeb4dd1

Signed-off-by: Tim Orling <tim.orling@konsulko.com>

Ecordonnier commented:
@moto-timo Thanks!
